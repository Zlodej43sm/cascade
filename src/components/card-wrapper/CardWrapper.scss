@use '../../styles/colors';
@use '../../styles/typography';
@use '../../styles/mixins';
@use '../../styles/sizes';

.container {
  @include mixins.themeBoxShadow(colors.$base-darkest);
  padding: sizes.$base-3 sizes.$base-3 sizes.$base-5;
  background-color: colors.$base-lightest;
  border-radius: 10px;
}

.header {
  display: flex;
  margin-bottom: sizes.$base-3;
}

.icon {
  font-size: 42px;
  margin-right: sizes.$base-3;
  flex: sizes.$base-4 0 0;
  text-align: center;
  position: relative;
  top: - sizes.$base;
}

.icon,
.subTitle {
  color: colors.$cascade-secondary;
}

.title {
  @include typography.title-md;
  color: colors.$cascade-primary;
}

.subTitle {
  @include typography.sub-title;
}

.content {
  overflow: auto;
}
