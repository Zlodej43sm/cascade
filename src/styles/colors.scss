$base-lightest: #fff;
$base-darkest: #000;
$base-dark: #3C3C3C;
$base-green: #8CBC5B;
$base-red: #FF5858;
$base-grey: #8C8C8C;

$grey-0: #FAFBFB;
$grey-10: #F2F2F2;
$grey-20: #DFDFDF;
$grey-30: #BFBFBF;

$cascade-primary: #3C3C3C;
$cascade-secondary: #594BDB;
