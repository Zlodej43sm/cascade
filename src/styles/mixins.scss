// Base theme styling
@mixin resetButtonStyles {
  border: none;
  overflow: visible;
  box-sizing: border-box;
  background-color: transparent;
  cursor: pointer;
}

@mixin themeBoxShadow($color) {
  box-shadow: 0 10px 30px rgba($color, 0.161);
}

// Theme elements
@mixin triangle($color) {
  content: '';
  margin-right: 8px;
  border-left: 14px solid transparent;
  border-right: 14px solid transparent;
  border-bottom: 24px solid $color;
}

@mixin crossMark($lineHeight, $lineWidth, $color) {
  position: relative;

  &::before, &::after {
    content: '';
    position: absolute;
    border-radius: 3px;
    height: $lineHeight;
    top: calc(50% - #{$lineHeight}/2);
    left: calc(50% - #{$lineWidth}/2);
    width: 2px;
    background-color: $color;
  }
  &:before {
    transform: rotate(45deg);
  }
  &:after {
    transform: rotate(-45deg);
  }
}
