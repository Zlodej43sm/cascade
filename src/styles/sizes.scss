// Theme spanning
$base: 8px;
$base-2: 2 * $base;
$base-3: 3 * $base;
$base-4: 4 * $base;
$base-5: 5 * $base;
$base-6: 6 * $base;
$base-12: 12 * $base;
$base-22: 22 * $base;

// Icons
$icon-md: $base-3;

// Forms
$textarea-dialog-height: $base-22;

// Buttons
$button-compact-min-width: $base-12;
$button-sm: $base-6;

