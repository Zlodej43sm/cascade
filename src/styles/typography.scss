@use 'fonts/poppins';
@use 'fonts/mulish';

$Font-Poppins: "Poppins", sans-serif;
$Font-Mulish: "Mulish", sans-serif;

@mixin poppins($size, $line, $weight: normal) {
  font-family: $Font-Poppins;
  font-size: $size;
  line-height: $line;
  font-weight: $weight;
}

@mixin mulish($size, $line, $weight: normal) {
  font-family: $Font-Mulish;
  font-size: $size;
  line-height: $line;
  font-weight: $weight;
}

// Heading
@mixin title-xl {
  @include poppins(36px, 54px, 600);
}

@mixin title-lg {
  @include poppins(24px, 36px, 500);
}

@mixin title-md {
  @include poppins(18px, 27px, 500);
}

@mixin sub-title {
  @include poppins(12px, 18px, 500);
}

// Buttons
@mixin button {
  @include poppins(14px, 30px, 500);
}

@mixin button-compact {
  @include poppins(16px, 24px, 400);
}
