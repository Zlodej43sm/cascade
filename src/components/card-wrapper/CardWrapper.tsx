import React from 'react';

import styles from './CardWrapper.scss';

export type CardWrapperProps = {
  icon?: React.ReactNode
  title?: string
  subTitle?: string
  children?: React.ReactNode
  width?: number | string
  height?: number | string
};

function CardWrapper({ icon, title, subTitle, children, width, height }: CardWrapperProps) {
  return (
    <section className={styles.container} style={{ width, height }} data-testid="card-wrapper-container">
      <header className={styles.header}>
        {icon && (
          <div className={styles.icon} data-testid="card-wrapper-icon">
            <span>{icon}</span>
          </div>
        )}
        <div>
          {title && (
            <div className={styles.title} data-testid="card-wrapper-title">
              {title}
            </div>
          )}
          {subTitle && (
            <div className={styles.subTitle} data-testid="card-wrapper-subtitle">
              {subTitle}
            </div>
          )}
        </div>
      </header>
      <div className={styles.content}>
        {children}
      </div>
    </section>
  );
}

export default CardWrapper;
