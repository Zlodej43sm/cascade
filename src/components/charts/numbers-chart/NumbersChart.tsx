import React from 'react';
import clsx from 'clsx';

import CardWrapper, { CardWrapperProps } from 'lib/components/card-wrapper/CardWrapper';

import styles from './NumberChart.scss';

export enum ChartTypes {
  Default = 'default',
  Grow = 'grow',
  Comparison = 'comparison',
}

export enum GrowTypes {
  Up = 'up',
  Down = 'down',
}

export type ChartProps = {
  value: string
  difference?: string
  unit?: string
  grow?: GrowTypes
  type?: ChartTypes
};

export function NumbersChart({
  value,
  difference,
  unit = '',
  type = ChartTypes.Default,
  grow = GrowTypes.Up,
}: ChartProps) {
  const containerClass = clsx(styles.container, styles[type], styles[grow]);
  const isComparison = type === ChartTypes.Comparison;
  return (
    <div className={containerClass} data-testid="number-chart">
      <span className={styles.value} data-testid="value">
        {unit}
        {value}
      </span>
      {isComparison && (
        <span className={styles.difference} data-testid="comparison-value">
            {difference}
        </span>
      )}
    </div>
  );
}

// solution for Ember usage
function NumbersChartCard(props: CardWrapperProps & ChartProps) {
  const {
    icon, title, subTitle, width, height, ...rest
  } = props;
  return (
    <CardWrapper {...{
      icon, title, subTitle, width, height,
    }}
    >
      <NumbersChart {...rest} />
    </CardWrapper>
  );
}

export default NumbersChartCard;
