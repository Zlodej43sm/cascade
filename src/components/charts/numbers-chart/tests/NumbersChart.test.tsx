import React from 'react';
import clsx from 'clsx';
import { render } from '@testing-library/react';

import { cardArguments } from 'lib/components/card-wrapper/__mocks__/CardWrapper.mock';
import { chartArguments } from '../__mocks__/NumbersChart.mock';
import NumbersChartCard, { NumbersChart } from '../NumbersChart';
import styles from '../NumbersChart.scss';

describe('NumbersChart', () => {
  const { defaultProps, growProps, growDownProps, differenceProps } = chartArguments;

  it('Should match snapshot NumbersChart', () => {
    const { asFragment } = render(<NumbersChart {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Should match snapshot for NumbersChartCard', () => {
    const { icon, title, subTitle } = cardArguments;
    const { asFragment } = render(<NumbersChartCard {...{
      ...defaultProps, icon, title, subTitle,
    }}
    />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Should render default type chart when type is not defined', () => {
    const expectedContainerClassName = clsx(styles.container, styles[growProps.grow]);
    const notDefinedTypeProps = { ...defaultProps, type: undefined };
    const { getByTestId } = render(<NumbersChart {...notDefinedTypeProps} />);
    expect(getByTestId('number-chart')).toHaveClass(expectedContainerClassName);
  });

  it('Should render grow down type chart', () => {
    const expectedContainerClassName = clsx(styles.container, styles[growDownProps.type], styles[growDownProps.grow]);
    const { getByTestId } = render(<NumbersChart {...growDownProps} />);
    expect(getByTestId('number-chart')).toHaveClass(expectedContainerClassName);
  });

  it('Should render comparison type chart', () => {
    const { getByTestId } = render(<NumbersChart {...differenceProps} />);
    expect(getByTestId('comparison-value')).toBeDefined();
  });
});
