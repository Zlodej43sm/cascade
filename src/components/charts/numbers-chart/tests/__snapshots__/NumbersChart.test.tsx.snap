// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumbersChart Should match snapshot NumbersChart 1`] = `
<DocumentFragment>
  <div
    class="container up"
    data-testid="number-chart"
  >
    <span
      class="value"
      data-testid="value"
    >
      11,458
    </span>
  </div>
</DocumentFragment>
`;

exports[`NumbersChart Should match snapshot for NumbersChartCard 1`] = `
<DocumentFragment>
  <section
    class="container"
    data-testid="card-wrapper-container"
  >
    <header
      class="header"
    >
      <div
        class="icon"
        data-testid="card-wrapper-icon"
      >
        <span>
          #
        </span>
      </div>
      <div>
        <div
          class="title"
          data-testid="card-wrapper-title"
        >
          FY21 Revenue
        </div>
        <div
          class="subTitle"
          data-testid="card-wrapper-subtitle"
        >
          As of today, in comparison with last month
        </div>
      </div>
    </header>
    <div
      class="content"
    >
      <div
        class="container up"
        data-testid="number-chart"
      >
        <span
          class="value"
          data-testid="value"
        >
          11,458
        </span>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
